add meetings 2021-09-22 notes (#263)
solid/authorization-panel: meetings/2021-09-22.md

* **Barath**: ultimately systems work well when they are pluralistic. we don't need to produce a single answer.. we can't forsee all future Use Cases.. talking about diverse world might be using this thing.. what i'd like to see is.. it is OK to see different specs.. as long as there is loose interop. and people can migrate... web works through coexistence.. 
* **Pavlik**: translating data grants into ACP rules. data grants give a way for auth agents or auth apps.. owned by different users.
* **Kjetil**: purpose of the issue i opened is to establish baseline interop. By putting in the protocol spec's the various operations we see coming out of this.. on the greater issue of diversity of work: there is more of a balancing act. Unecessary divergence is  likely to hurt interop. on the other hand, being too restrictive may stop people from being creative and that is also bad. 
* **Sarven**: +1 protocol spec as baseline.
* **Sarven**: why isn't the data grants being translated to WAC?
* **Pavlik**: EricP were looking into translating to WAC. wac-ucr had list of limitation found in that process.

> * **Sarven**: WAC is called basic access control ontology, it should be able to do a lot of stuff for you. There is long list of things the can also be done which it doesn't do - but it can via extensions.. or updating the spec.
> * ...: This didn't stop ACP to be considered, unless there are fundamental issues, we can't do greenfielding on everything.

Reviewer: ```suggestion
* **Sarven**: WAC.. ns/auth/acl is basic access control ontology, it should be able to do a lot of stuff for you. There is long list of things the can also be done which it doesn't do - but it can via extensions.. or updating the spec.
* ...: (*The scribe did not transcribe the context for this message but it was a continuation to above; while there is existing work, new work is welcome. the response is in context of operations, access controls, and making fundamental changes to the Solid Protocol re "greenfielding" as mentioned in the comments of the topic issue*) This didn't stop ACP to be considered, unless there are fundamental issues, we can't do greenfielding on everything.
```

* **Kjetil**: Can we agree that Read and Write can be top level classes. We could reach consensus already.
* **Sarven**: yes. they're class of operations. is there no consensus on that?
* **Kjetil**: Justin mentioned Write could be deprecated.. but that's not the case for a lot of social groups. Idea of having RW as upper classes .. can we have consensus on that.
* **Henry**: yes.. +1 for Read and Write as top level operations. That fits perfectly  with lenses. It is mind-boggling to think that this simple concept is behind so much (OO programming, Dyanamic systems, ...)
* **Sarven**: yes.
* **Pavlik**: WAC's RW or ?
* **Henry**: generic.
* **Sarven**: RW *operations*

> * **Henry**: Rread is observation with no change of state, Write is transformation of state.

Reviewer 2: ```suggestion
* **Henry**: Read is observation with no change of state, Write is transformation of state.
```

* **Pavlik**: 
* **Kjetil**: we're also talking about modify, create, delete.. and so on. how many operations will we bring in to the specification. Also we have to consider how the HTTP protocol sees them. There are a lot of things there to consider.
* **Pavlik**: I find it confusing to call both operation and access mode 'Read', 'Write'.
* **Henry**: Read is an observation function on a state S to some representation A
```scala
   read: S -> A
```
Write is a function that gives a representation A and a state and changes it to a new state
```scala
   write: S × A -> S 
  ```
* **Sarven**: ...
* **Sarven**: We can introduce new access model to Delete, it seems to just substitute variable name but end result seems the same.
* **Henry**: I think Justin should try to write how this example would work with WAC, and show what Delete would give us.
* **Sarven**: too much intelectualising not enough visible implementations.
* **Henry**: I will write client libraries next month, so I can try out these ideas.
* **Pavlik**: re Read operation on the resource.. ACL seems the nuance of e.g. Read mode.

> * **Matthieu**: wondering about UCs and think Henry is right.. Write on. !!something missing here 

Reviewer 2: ```suggestion
* **Matthieu**: Even though writing a good UC in order to understand which specific modes are needed is challenging, I think it is the right idea. 
```

> * **Sarven**: we have issue about error codes

Author: @matthieubosquet there is something missing here. Not sure what it was.

Reviewer 2: Suggested something that I think sumarizes what I thought at the time.
